### ISSUE TITLE
Save Pk in code or/and in repository production

### ISSUE THREAD

**Issue by londumas** [igmhub/SaclayMocks#18]
Save P(k) in code or/and in repository production. Where is it saved for the moment?

**Comment by londumas** [comment]
We also need to save the intup parameters of CAMB and the parameters (Omega_m) used in the box to transform z to r

**Comment by TEtourneau** [comment]
The P(k) we use is store here:
/global/u1/t/tetourne/Git/LyaMocks/bin/data/PlanckDR12.fits 
I think this is the one used in picca.
All the cosmological parameters are store in the file .../py/LyaMocks/Constant.py

```
h = 0.6731
omega_M_0 = 0.31457
omega_b_0 = 0.045  # only use in run_camb.py
omega_lambda_0 = 0.68543
omega_k_0 = 0.0
ns = 0.96  # only used in run_camb.py
```

**Comment by londumas** [comment]
@TEtourneau, perfect, let's store them also to the gitlab repository. Would that be ok?

**Comment by TEtourneau** [comment]
Everything is already on the gitlab repository:
the pk is in bin/data/
and the Constant.py library in py/LyaMocks/Constant.py

I'm working on cleaning up the git repository, and transfering everything on github.

**Comment by londumas** [comment]
@TEtourneau, did you redrive it yourself from CMB? I don't see the config file, it is important to keep it.

**Comment by TEtourneau** [comment]
No, I don't think so. I think it is the power spectrum used in picca.

**Comment by jmarclegoff** [comment]
@londumas, yes, indeed I think I remember that you asked us to use the picca PlanckDR12.fits instead of producing it ourselves. However the file we now have is different from the one in https://github.com/igmhub/picca/tree/master/py/picca/fitter2/models/PlanckDR12
May be you changed it meanwhile, so our file might be identical to an older version of the picca file.

**Comment by londumas** [comment]
@TEtourneau, in the London mocks they store the redshift-distance relation in the HDU of the `master.fits` file. I think it would be good to do the same. They have  `COSMO_COL` for the big cells and `COSMO_EXP` for the small ones. What do you think? The P(k) could be actually store also in another HDU of the `master.fits` file. This file would be then a "all you can eat buffet".

**Comment by TEtourneau** [comment]
I don't get why you are talking about cosmology of big and small cells. Could it be different ?
If not, we only need 1 HDU to give the distance relations.
So, to sum up, we need:
- 1 HDU with the QSO catalog, with cosmo parameters in the header
- 1 HDU with the distance relations
- 1 HDU with the input power spectrum
is that it ?

**Comment by TEtourneau** [comment]
Ah, and do you know what are the different fields in the distance relation hdu ?
R: comobile distance ? (in Mpc/h?)
Z: redshift ?
D: growth factor ?
V: ?

**Comment by londumas** [comment]
@TEtourneau, I asked exactly the same question to @jfarr03 there https://github.com/igmhub/LyaCoLoRe/issues/62.
If you use the same cosmology, then I would simply have only one `COSMODIS` with `DC` the comoving distance, `Z` the redshift `G` the growth factor, and I don't know what else; maybe the growth rate `F`. You know better what we need to keep, and how best to name them.
 I don't know what are the meaning of all the field they have in LyaCoLoRe.

And indeed, I would do:
 - 1 HDU with the QSO catalog, with cosmo parameters in the header
 - 1 HDU with the distance relations
 - 1 HDU with the input power spectrum

how does that sound @TEtourneau?

**Comment by TEtourneau** [comment]
Sorry, my question wasn't clear. I was just wondering the meaning of the 4 different fields.

Sure ! Let's go for these 3 HDU.
For the moment, I'll just store the redshift Z, the growth factor D and the comoving distance R.

**Comment by jmarclegoff** [comment]
I think we should discuss with James and Andreu to agree on what should be put and on the format, to avoid having different format. This discussion concerns both LyaColoRe and SaclayMocks, can we open a common issue, or add @jfarr03 and @andreufont to this discussion ?

**Comment by londumas** [comment]
@TEtourneau, is that fixed? Did you store both the P(k) and the CAMB .ini file?

**Comment by TEtourneau** [comment]
Sorry, I forgot to do it for the 10 realisations, but I can do it now (master file is really fast to generate).
About the CAMB .ini file, I don't have it, since the power spectrum I'm using is the one of picca.

I'll put in the different master files the input power spectrum, and the distance relations.

**Comment by londumas** [comment]
@TEtourneau, perfect, let's do that for the 10 master files.
For the .ini file and the P(k), let's have you copy both from https://github.com/igmhub/picca/tree/master/py/picca/fitter2/models/PlanckDR16 into the repository, so that we have exactly the same. Would that be ok @TEtourneau ?

**Comment by TEtourneau** [comment]
@londumas What do you mean ? 
For the next realisations ?
For the 10 realisations, I used a power spectrum you gave me some time ago, it's named PlanckDR12.fits, if that's helping you.

**Comment by londumas** [comment]
@TEtourneau, the power spectrum you used was generated using the following CAMB .ini: `py/picca/fitter2/models/PlanckDR12/PlanckDR12_kmax_params.ini`. However this file can not be used for new version of CAMB, because of the way the neutrino were set. The following file gives exactly the same P(k): `py/picca/fitter2/models/PlanckDR16/PlanckDR16.ini`.
So let's store this .ini file into SaclayMocks, so that we can know how the P(k) was generated.
Does that make sense?
P.S. could you be a bit more verbose than "Bug fix" in you commits? It will be useful in the future to track what changed.

**Comment by TEtourneau** [comment]
Ok, thanks for the explanations, I'll do it.
Ok for the commits ! :)
